opentracing/tracer.py

```diff
@@ -37,16 +39,72 @@ class Tracer(object):
 
     _supported_formats = [Format.TEXT_MAP, Format.BINARY, Format.HTTP_HEADERS]
 
-    def __init__(self):
+    def __init__(self, scope_manager=ScopeManager()):
+        self._scope_manager = scope_manager
         self._noop_span_context = SpanContext()
         self._noop_span = Span(tracer=self, context=self._noop_span_context)
+        self._noop_scope = Scope(self._scope_manager, self._noop_span)
+
+    @property
+    def scope_manager(self):
+        """ScopeManager accessor"""
+        return self._scope_manager
+
+    def start_active_span(self,
+                          operation_name=None,
+                          child_of=None,
+                          references=None,
+                          tags=None,
+                          start_time=None,
+                          ignore_active_span=False,
+                          finish_on_close=False):
+        """Returns a newly started and activated `Scope`.
+        Returned `Scope` supports with-statement contexts. For example:
+
+            with tracer.start_active_span('...') as scope:
+                scope.span.set_tag('http.method', 'GET')
+                do_some_work()
+            # Span is *not* finished automatically outside the `Scope` `with`.
+
+        It's also possible to finish the `Span` when the `Scope` context
+        expires:
+
+            with tracer.start_active_span('...',
```

Comment:
is there a reason the finish_on_close=True is not the default case? because that is what 99% of the use case will look like. The default value should be based on the default use case for active spans.

Reply:
totally agree, I think that if you create an active span it should finish automatically. In that way snippets like the following works out of the box:

```python
with tracer.start_active_span('web.request'):
  # do something
```

Reply:
There a discussion about that during the Java API design: https://github.com/opentracing/opentracing-java/issues/189#issuecomment-334730068 (pretty much the long debate + eventual PR was between instrumentation vs user code). The general opinion was going in the direction of `finishSpanOnClose=false` till we ended up removing the default value for it ;)

This (sadly) can't be done for the Python API here, as `start_active_span` takes all parameters as optional (even `operation_name`, as stated in the description).

That being said, I'd like to hear what other people think, and change to to `True` if needed ;)

Reply:
For the sake of keeping momentum, I'd be for:

1. Keeping things as they are (`finishSpanOnClose=false`)
2. Adding some comments to the method to help implementers make a decision on `finishSpanOnClose`.

IMO, this is most likely to bite end-users that are custom instrumenting their own code and aren't doing tracing as their day job. It's less likely to impact `contrib` repos as those are more likely to have a set of eyes on them. With OT early in its lifecycle, it feels like more of the work is on `contrib` than end users anyway, so this may be the type of decision that's OK to punt on for a while.

Reply:
@itsderek23 Hey, thanks for the feedback ;) So, yes, one of the arguments in favor of `finishSpanOnClose=false` was the fact most people using this will be writing library instrumentation, rather than application code (and usually, OT is used there through a middleware-alike layer, so often a `Span` is not automatically closed, but activated and passed around).

Of course, there's also the argument that @beberlei and @palazzem do... so lets see and wait for more feedback. Nevertheless, I'd expect this moving (momentum, etc) right this week ;)